CONTRIBUTING.md

```diff
@@ -0,0 +1,276 @@
+    return _newfrom + (_newto - _newfrom) * _amount;
+}
+```
+
+**NO:**
+
+```gml
+function relerp(oldfrom, oldto, oldvalue, newfrom, newto) {
+    if (oldto == oldfrom) {
+        throw "relerp() old range has zero width.";
+    }
+    var amount = (oldvalue - oldfrom) / (oldto - oldfrom);
+    return newfrom + (newto - newfrom) * amount;
+}
+```
+
+### Toolbox Script Assets
+
+- **Use `utils_` as the prefix, and PascalCase for the rest of the name.** Example: `utils_CommunityToolboxScriptCategory`
+
+### Other assets
+
+- **Use lowercase+underscore asset type prefixes (e.g. `spr_`, `obj_`, `rm_`), followed by PascalCase for the rest of the name.** Example: `rm_Home`
+
+## Code Style
+
+### Identations
+
+- **Use 4 space characters for an indentation level, without tab characters.**
+
+### Braces
+
+**Use [K&R bracing style](https://en.wikipedia.org/wiki/Indentation_style#K&R_style):**
+
+- Opening code block braces are in the same line as block definition (including functions)
+- Closing braces are in their own line, at the same identation level as the original block-starting expression.
+- Prefer using braces whenever applicable, even for one-liners.
+
+Example:
+
+```gml
+function some_function(_arg1, _arg2, _arg3) {
+    var _some_arg = 123;
+    if (_arg1 == _arg2) {
+        // do some thing
+    } else if (_arg1 == _arg3) {
+        // do other thing
+    } else {
+        // do yet another thing
+    }
+}
+```
+
+- If braces are omitted in an `if` block, putting the condition in the first line and the body of the `if` in the second line.
+
+```gml
+function other_function(_arg1) {
+    if (is_undefined(_arg1))
+        return undefined;
+
+    // do the rest
+}
+```
+
+- If there's an `if`-`else` block, then the braces should be consistently applied or consistently omitted.
+
+**YES:**
+
+```gml
+if (_some_condition)
+    return 1;
+else if (_other_condition)
+    return 2;
+else
+    return 3;
+```
+
+**YES:**
+
+```gml
+if (_some_condition) {
+    return 1;
+} else if (_other_condition) {
+    return 2;
+} else {
+    return 3;
+}
+```
+
+**NO:**
+
+```gml
+if (_some_condition)
+    return 1;
+else if (_other_condition)
+    return 2;
+else {
+    show_debug_message("Hello, world!");
+    return 3;
+}
+```
+
+- Two separate `if` blocks (not part of the same `if`/`else` ladder) must be at least one line apart.
+
+**YES:**
+
+```gml
+if (_some_condition)
+    show_debug_message("AAA");
+
+if (_other_condition)
+    show_debug_message("BBB");
+```
+
+**YES:** (same `if`/`else` ladder)
+
+```gml
+if (_some_condition)
+    show_debug_message("AAA");
+else if (_other_condition)
+    show_debug_message("BBB");
+```
+
+**NO:** (not the same `if`/`else` ladder)
+
+```gml
+if (_some_condition)
+    show_debug_message("AAA");
+if (_other_condition)
+    show_debug_message("BBB");
+```
+
+### Semicolons
+
+- **Always put a semicolon after regular statements.**
+
+**YES:**
+
+```gml
+//OK
+show_debug_message("Hello, world!");
```

Comment:
```suggestion
show_debug_message("Hello, world!");
```
I saw that you implemented most suggestions, just pointing out this runaway OK comment. Otherwise, we're on a good track! ^^